File: bms/go/geo/geo_s1_ChordAngle_isValid/geo_s1_ChordAngle_isValid.go
package test

import (
	"math"
)

// var ()

const (
	geo_s1_maxLength2 = 4.0
)

func (c geo_s1_ChordAngle) isValid() bool {
	return (c >= 0 && c <= geo_s1_maxLength2) || c.isSpecial()
}

func (c geo_s1_ChordAngle) isSpecial() bool {
	return c < 0 || c.IsInfinity()
}

func (c geo_s1_ChordAngle) IsInfinity() bool {
	return math.IsInf(float64(c), 1)
}

type geo_s1_ChordAngle float64

File: translations_only/fail/go/geo/geo_s1_ChordAngle_isValid/claude3/CAPR/geo_s1_ChordAngle_isValid.rs

use std::f64;

const GEO_S1_MAX_LENGTH2: f64 = 4.0;

fn is_valid(c: GeoS1ChordAngle) -> bool {
    (c >= 0.0 && c <= GEO_S1_MAX_LENGTH2) || is_special(c)
}

fn is_special(c: GeoS1ChordAngle) -> bool {
    c < 0.0 || is_infinity(c)
}

fn is_infinity(c: GeoS1ChordAngle) -> bool {
    f64::is_infinite(c)
}

type GeoS1ChordAngle = f64;
